smart_contracts.rst

```diff
@@ -317,10 +268,44 @@ Allows the participants to cooperate and provide both of their balances and sign
 
     Can be called by a third party as long as both participants provide their signatures.
 
+
+**Unlock lock**
+
+Unlocks all pending transfers by providing the entire merkle tree of pending transfers data. The merkle tree is used to calculate the merkle root, which must be the same as the ``locksroot`` provided in the latest balance proof.
+
+::
+
+    function unlock(
+        uint channel_identifier,
+        address participant,
+        address partner,
+        bytes merkle_tree)
+        public
+
+::
+
+    event ChannelUnlocked(uint256 channel_identifier, address payer_participant, uint256 unlocked_amount, uint256 returned_tokens);
+
+- ``channel_identifier``: Channel identifier assigned by the current contract.
+- ``participant``: Ethereum address of the channel participant who will receive the unlocked tokens that correspond to the pending transfers that have a revealed secret.
+- ``partner``: Ethereum address of the channel participant that pays the amount of tokens that correspond to the pending transfers that have a revealed secret. This address will receive the rest of the tokens that correspond to the pending transfers that have not finalized and do not have a revelead secret.
+- ``merkle_tree``: The entire merkle tree of pending transfers. It contains tightly packed data for each transfer, consisting in ``expiration_block``, ``locked_amount``, ``secrethash``.
+- ``expiration_block``: The absolute block number at which the lock expires.
+- ``locked_amount``: The number of tokens being transferred from ``partner`` to ``participant`` in a pending transfer.
+- ``secrethash``: A hashed secret, ``sha3_keccack(secret)``.
+- ``payer_participant``: Ethereum address of the channel participant whose ``transferred_amount`` will be increased.
```

Comment:
Changing this because at the moment it is a combination of the both the old and new approaches. So, I am removing `payer_participant` because it is useless and replace it in the text with `partner`.

`event ChannelUnlocked(uint256 channel_identifier, address participant, uint256 unlocked_amount, uint256 returned_tokens);`